internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -835,25 +864,49 @@ func processorsForProcessMetrics(binaryName, unitID, namespace, dataset string,
 }
 
 // processorsForBeatsStream returns the processors used for metric streams in the beats input.
-func processorsForBeatsStream(binaryName, unitID, namespace, dataset string, agentInfo info.Agent) []any {
-	return []any{
+func processorsForBeatsStream(
+	binaryName, unitID, namespace, dataset string,
+	agentInfo info.Agent,
+	runtimeManager component.RuntimeManager,
+) []any {
+	processors := []any{
 		addDataStreamFieldsProcessor(dataset, namespace),
 		addEventFieldsProcessor(dataset),
 		addElasticAgentFieldsProcessor(binaryName, agentInfo),
 		addAgentFieldsProcessor(agentInfo.AgentID()),
 		addComponentFieldsProcessor(binaryName, unitID),
 	}
+	if runtimeManager == component.OtelRuntimeManager { // we don't want process metrics for beats receivers
```

Comment:
These metrics are duplicated, but the UI may still want them.

We need to take a look at the [elastic_agent](https://github.com/elastic/integrations/blob/main/packages/elastic_agent) integration and make sure things still work.

What should happen is that `metrics-elastic_agent-*` is enough to calculate this and you no longer need to also account for `metrics-elastic-agent.$beat-*` in queries and dashboards. It should be possible for this to work but we need to explicitly check it.

Reply:
The Fleet CPU and Metric aggregations are going to be querying into the per beat indexes I think, we need to make sure we didn't break that https://github.com/elastic/elastic-agent/blob/main/docs/cpu-metrics-in-fleet.md

Reply:
At a glance, the Fleet aggregations don't care about the index, so that part should be fine. I'm not completely sure about the beats-specific parts of the agent integration - I'll have a look at those.

I'd be inclined to file a follow-up issue and investigate there. Modifying what exactly these events contain isn't particularly difficult, we can always fix it later if it turns out the process metrics need to exist after all.

Reply:
I tested this change (alongside the follow-up change enabling the use of beats receivers for self-monitoring) with Fleet, and didn't see any problems related to CPU and memory metrics. Both on the agent list, and in the Elastic Agent integration dashboard, these values were shown correctly.

I did see some issues, but these are related to beats internal metrics and are already tracked in https://github.com/elastic/beats/issues/44030.